psl/psl-core/src/validate/validation_pipeline/validations/indexes.rs

```diff
@@ -386,20 +386,23 @@ pub(crate) fn opclasses_are_not_allowed_with_other_than_normal_indices(index: In
     }
 }
 
-pub(crate) fn composite_types_are_not_allowed_in_index(index: IndexWalker<'_>, ctx: &mut Context<'_>) {
-    for field in index.fields() {
-        if field.scalar_field_type().as_composite_type().is_some() {
-            let message = format!(
-                "Indexes can only contain scalar attributes. Please remove {:?} from the argument list of the indexes.",
-                field.name()
-            );
-            ctx.push_error(DatamodelError::new_attribute_validation_error(
-                &message,
-                index.attribute_name(),
-                index.ast_attribute().span,
-            ));
-            return;
-        }
+pub(crate) fn composite_type_in_compound_index(index: IndexWalker<'_>, ctx: &mut Context<'_>) {
+    let composite_type = index
+        .fields()
+        .find(|f| f.scalar_field_type().as_composite_type().is_some());
+
+    if index.fields().len() > 1 && composite_type.is_some() {
+        
+        let message = format!(
+            "Prisma does not currently support composite types in compound indices. Please remove {:?} from the index.",
+            composite_type.unwrap().name()
+        );
+        ctx.push_error(DatamodelError::new_attribute_validation_error(
+            &message,
+            index.attribute_name(),
+            index.ast_attribute().span,
+        ));
```

Comment:
This needs a little bit more work.
Unsure if this should be an _error_ or simply a _warning_.
Error holds a little more gravitas, and currently attempting to use this functionality causes the query-engine to panic on `generate`. However, it does work if you `--skip-generate`, so if you're just using Prisma as a schema management system, this is still a regression in functionality which makes me consider downgrading this to a warning.